12-offer-encoding.md

```diff
@@ -471,6 +635,34 @@ The writer:
         - MUST set `invreq_quantity` less than or equal to `offer_quantity_max`.
     - otherwise:
       - MUST NOT set `invreq_quantity`
+    - if `offer_recurrence_optional` or `offer_recurrence_compulsory` are present:
+      - for the initial request:
+        - MUST use a unique `invreq_payer_id`.
+        - MUST set `invreq_recurrence_counter` `counter` to 0.
+        - MUST NOT set `invreq_recurrence_cancel`.
+      - for any successive requests:
+        - MUST use the same `invreq_payer_id` as the initial request.
+        - MUST set `invreq_recurrence_counter` `counter` to one greater than the highest-paid invoice.
+        - MAY set `invreq_recurrence_cancel`.
+      - if `offer_recurrence_base` is present:
+        - MUST include `invreq_recurrence_start`
+        - MUST set `period_offset` to the period the sender wants for the initial request
```

Comment:
Is the period offset from `offer_recurrence_base`? If so, could we clarify that? In general, would it make sense to have a dedicated section (like the recurrence section) defining the exact meaning of `invreq_recurrence_counter`, `invreq_recurrence_start`, and `invreq_recurrence_cancel`?

Reply:
The offer_recurrence_base says "Period 0 always starts Feb 1st" for example.  Say it's monthly.

The user wants to start in August.  So, she sets `invreq_recurrence_start`'s `period_offset` to 6.  But her first invoice request, *as always* is for `invreq_recurrence_counter` 0.  That means it's trivial for the receiver to spot the first in a series of invoice requests....

I prefer to avoid nomenclature sections if we can make it clear in-situ.  Nobody (else!) ever reads them.

Reply:
Okay, that clears it up! On a related note, do you think it would make sense to rename it to `period_index_offset`? Since it refers to the index rather than an actual time offset, it would better align with the `max_period_index` naming we landed on earlier, and could help avoid confusion.